src/main/java/soot/Hierarchy.java

```diff
@@ -202,70 +202,103 @@ protected void checkState() {
 		return l;
 	}
 
-	/** Returns a list of superclasses of c, including itself. */
-	public List<SootClass> getSuperclassesOfIncluding(SootClass c) {
-		c.checkLevel(SootClass.HIERARCHY);
-		List<SootClass> l = getSuperclassesOf(c);
-		ArrayList<SootClass> al = new ArrayList<SootClass>();
-		al.add(c);
-		al.addAll(l);
-		return Collections.unmodifiableList(al);
+	/**
+	 * Returns a list of superclasses of {@code sootClass}, including itself.
+	 *
+	 * @param sootClass the <strong>class</strong> of which superclasses will be taken. Must not be {@code null} or
+	 *                  interface
+	 * @return list of superclasses, including itself
+	 * @throws IllegalArgumentException when passed class is an interface
+	 * @throws NullPointerException     when passed argument is {@code null}
+	 */
+	public List<SootClass> getSuperclassesOfIncluding(SootClass sootClass) {
+		final List<SootClass> superclasses = getSuperclassesOf(sootClass);
+
+		final List<SootClass> result = new ArrayList<>(superclasses.size() + 1);
+		result.add(sootClass);
+		result.addAll(superclasses);
+
+		return Collections.unmodifiableList(result);
 	}
 
-	/** Returns a list of strict superclasses of c, starting with c's parent. */
-	public List<SootClass> getSuperclassesOf(SootClass c) {
-		c.checkLevel(SootClass.HIERARCHY);
-		if (c.isInterface())
-			throw new RuntimeException("class needed!");
+	/**
+	 * Returns a list of <strong>direct</strong> superclasses of passed class in reverse order,
+	 * starting with its parent.
+	 *
+	 * @param sootClass the <strong>class</strong> of which superclasses will be taken. Must not be {@code null} or
+	 *                  interface
+	 * @return list of superclasses
+	 * @throws IllegalArgumentException when passed class is an interface
+	 * @throws NullPointerException     when passed argument is {@code null}
+	 */
+	public List<SootClass> getSuperclassesOf(SootClass sootClass) {
+		sootClass.checkLevel(SootClass.HIERARCHY);
+		if (sootClass.isInterface()) {
+			throw new IllegalArgumentException(sootClass.getName() + " is an interface, but class is expected");
+		}
 
 		checkState();
 
-		ArrayList<SootClass> l = new ArrayList<SootClass>();
-		SootClass cl = c;
+		final List<SootClass> superclasses = new ArrayList<>();
 
-		while (cl.hasSuperclass()) {
-			l.add(cl.getSuperclass());
-			cl = cl.getSuperclass();
+		SootClass current = sootClass;
+		while (current.hasSuperclass()) {
+			superclasses.add(current.getSuperclass());
+			current = current.getSuperclass();
 		}
 
-		return Collections.unmodifiableList(l);
+		return Collections.unmodifiableList(superclasses);
 	}
 
-	/** Returns a list of subinterfaces of c, including itself. */
-	public List<SootClass> getSubinterfacesOfIncluding(SootClass c) {
-		c.checkLevel(SootClass.HIERARCHY);
```

Comment:
Why have you removed the check for the resolving level here while still preserving it everywhere else? This seems to be an issue to me.